### ISSUE TITLE
How can I gain feature vectors or other middle outputs?

### ISSUE THREAD

**Issue by wengzejia1** [deepmind/deepmind-research#6]
Hello, I am trying to do some work on the "Unsupervised Adversarial Training" cifar10 pre-train model. But I meet some problem that I do not know how to get the middle outputs besides the final logits output. Can you give me some advice on how to achieve my goal? 

Thanks!

**Comment by juesato** [comment]
Hi, thanks for the interest. We don't directly expose the internal activations of the network via the TF-Hub interface, but you can do this in an adhoc way via `get_tensor_by_name`.

You might find these other threads helpful, which also discuss how to retrieve internal activations from TF-Hub modules.
https://github.com/tensorflow/hub/issues/290
https://github.com/tensorflow/hub/issues/396

For figuring out the names of each Tensor, I would use some combination of (1) inspecting the names of Tensors and Variables and (2) checking `op.inputs` and `op.outputs`.
As a final helpful tip, the model is also divided into Variable scopes based on each ResNet "block" and then "layers" within each block (the same way the architecture is described in https://arxiv.org/abs/1605.07146), so if e.g. you're trying to retrieve the activation after each residual block, this should basically involve changing a single number e.g. '/long/variable/scope/path/resnet_block_1/MatMul:0' to '/long/variable/scope/path/resnet_block_2/MatMul:0'

**Comment by diegolascasas** [comment]
Closing this issue, feel free to reopen it if you have further questions.

**Comment by wmcnally** [comment]
Are there any plans to improve the interface of TF Hub to facilitate getting outputs of intermediate layers? TF Hub seems like a great resource (lots of pre-trained models) but without being able to interface intermediate layers the applications are severely limited (e.g., I cannot easily build a custom model). I'm currently using tf.keras.applications -- I would like to use some of the architectures in TF Hub in the same way.
